code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -459,17 +462,62 @@
 	return ..()
 
 /mob/living/simple_animal/bot/screwdriver_act(mob/living/user, obj/item/I)
+	. = TRUE //Must be true or we attempt to stab the bot | Sets standard return value for this code block
+
 	if(user.a_intent == INTENT_HARM)
 		return ..()
 	if(locked)
-		to_chat(user, "<span class='warning'>The maintenance panel is locked.</span>")
-		return TRUE // Must be true or we attempt to stab the bot
+		to_chat(user, "<span class='warning'>The interface is locked.</span>")
+		return
 
-	open = !open
 	I.play_tool_sound(src)
+	open = !open
 	to_chat(user, "<span class='notice'>The maintenance panel is now [open ? "opened" : "closed"].</span>")
+	return
+
+/mob/living/simple_animal/bot/wrench_act(mob/living/user, obj/item/I)
+	. = TRUE //Must be true or we attempt to stab the bot | Sets standard return value for this code block
+
+	if(user.a_intent != INTENT_HELP)
+		return..()
+	if(locked) //ID-Lock
+		to_chat(user, "<span class='warning'>The interface is locked.</span>")
+		return
+	if(!open) //Screwdriver panel
+		to_chat(user, "<span class='warning'>The maintenance panel is screwed in place.</span>")
+		return
+	if((bot_type != SEC_BOT) && (bot_type != GRIEF_BOT)) //Should only affect security bots and griefsky
+		to_chat(user, "<span class='warning'>The internal components cannot be wrenched more securely.</span>")
+		return
+
+	I.play_tool_sound(src)
+	wrenched = !wrenched
+	to_chat(user, "<span class='notice'>The internal components are now [wrenched ? "wrenched" : "unwrenched"].</span>")
 	return TRUE
 
+// Override me please!
+/mob/living/simple_animal/bot/proc/disassemble()
+	CRASH("disassemble() not overridden on [type]")
+
+/mob/living/simple_animal/bot/multitool_act(mob/living/user, obj/item/I)
+	. = TRUE //Must be true or we attempt to stab the bot | Sets standard return value for this code block
+
+	if(user.a_intent != INTENT_HELP)
+		return ..()
+	if(locked) //ID-Lock
+		to_chat(user, "<span class='warning'>The interface is locked.</span>")
+		return
+	if(!open) //Screwdriver panel
+		to_chat(user, "<span class='warning'>The maintenance panel is screwed in place.</span>")
+		return
+	if(((bot_type == SEC_BOT) || (bot_type == GRIEF_BOT)) && (wrenched == TRUE)) //Should only affect security bots and griefsky that have their components wrenched
+		to_chat(user, "<span class='warning'>The internal components are wrenched in place.</span>")
+		return
+
+	I.play_tool_sound(src)
+	to_chat(user, "<span class='notice'>You disassemble [src].</span>")
+	disassemble()
+	return
```

Comment:
```suggestion
```
unnecessary hanging return

Reply:
Just to be sure, was the return unnecessary because a proc was called and that said proc had it's own returns and end arguments ? Or would the above code blocks (such as `wrench_act` and `screwdriver_act`) not need their final returns as well ?

Reply:
once the last line of a proc is reached and executed, the proc will automatically return on it's own. I.g. all procs inherently run `return` => `return .` as the very last thing they do if there is not a provided return command already.

If all you are doing is manually running `return` at the end of the proc, you're just adding uneccesary code as BYOND is going to do that anyway. Hence the name, "hanging return"

Reply:
Okay! Thank you for the explanation